### PR TITLE
Pivotal ID # 187229373: Backend using Java 17

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/187229373
- Updated to use Java 17

### CODE REVIEW

File: client/ftp-webclient/src/test/kotlin/ebi/ac/uk/ftp/FtpClientTest.kt [hunk 1]
```diff
@@ -18,13 +17,9 @@ class FtpClientTest {
...
         )
 
-    @BeforeEach
-    fun beforeEach() {
-        testInstance.deleteFile(Paths.get(""))
-    }
-
     @RetryingTest(TEST_RETRY)
     fun uploadFileInFolder() {
+        testInstance.deleteFile(HOME)
```
Comment (Juan-EBI) [inline client/ftp-webclient/src/test/kotlin/ebi/ac/uk/ftp/FtpClientTest.kt:22]: moving this one to here as @RetryingTest does not work when is in `beforeEach`

File: submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/pagetab/PageTabService.kt [hunk 2]
```diff
@@ -81,12 +82,12 @@ class PageTabService(
...
         version: Int,
     ): File {
-        val now = LocalDate.now()
+        val now = ZonedDateTime.now(clock)
         val path =
             baseTempFolder
-                .resolve(now.get(ChronoField.YEAR).toString())
-                .resolve(now.get(ChronoField.MONTH_OF_YEAR).toString())
-                .resolve(now.get(ChronoField.DAY_OF_MONTH).toString())
+                .resolve(now.year.toString())
```
Comment (jhoanmanuelms) [inline submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/pagetab/PageTabService.kt:88]: 👍

File: submission/submission-handlers/src/test/kotlin/ac/uk/ebi/biostd/handlers/exception/CustomErrorHandlerTest.kt [hunk 3]
```diff
@@ -28,11 +26,11 @@ class CustomErrorHandlerTest(
     fun `handle error`() {
         val alertSlot = slot<Alert>()
         val expectedAlert = Alert(SYSTEM_NAME, HANDLERS_SUBSYSTEM, "$ERROR_MESSAGE: the error message")
+        val expection = RuntimeException("the error message")
```
Comment (jhoanmanuelms) [inline submission/submission-handlers/src/test/kotlin/ac/uk/ebi/biostd/handlers/exception/CustomErrorHandlerTest.kt:29]: typo: exception

**Review by jhoanmanuelms — APPROVED** [review]
LGTM. Nice upgrade, one minor comment

**Pull request merged**